syntax = "proto3";

package idm;

service IDMService {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc ValidateToken (TokenRequest) returns (TokenValidationResponse);
  rpc Logout (TokenRequest) returns (GenericResponse);
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message LoginResponse {
  string token = 1;
}

message TokenRequest {
  string token = 1;
}

message TokenValidationResponse {
  bool valid = 1;
  int32 user_id = 2;
  string role = 3;
  string email = 4;
  string message = 5;
}

message GenericResponse {
  bool success = 1;
  string message = 2;
}
